polkadot/node/network/collator-protocol/src/validator_side/collation.rs

```diff
@@ -18,16 +18,25 @@
 //!
 //! Usually a path of collations is as follows:
 //!    1. First, collation must be advertised by collator.
-//!    2. If the advertisement was accepted, it's queued for fetch (per relay parent).
-//!    3. Once it's requested, the collation is said to be Pending.
-//!    4. Pending collation becomes Fetched once received, we send it to backing for validation.
-//!    5. If it turns to be invalid or async backing allows seconding another candidate, carry on
+//!    2. The validator inspects the claim queue and decides if the collation should be fetched
+//!       based on the entries there. A parachain can't have more fetched collations than the
+//!       entries in the claim queue at a specific relay parent. When calculating this limit the
+//!       validator counts all advertisements within its view not just at the relay parent.
+//!    3. If the advertisement was accepted, it's queued for fetch (per relay parent).
+//!    4. Once it's requested, the collation is said to be Pending.
+//!    5. Pending collation becomes Fetched once received, we send it to backing for validation.
```

Comment:
Don't we treat it as Pending at this step as well?

Reply:
From collation fairness point of view - yes. But generally speaking the module differs between collations being fetched (`CollationStatus::Fetching`) and collations pending validation (`CollationStatus::PendingValidation`).

I've rephrased the comment to avoid this confusion.